### PR TITLE
Refactoring how Adapters are managed.

### PR DESCRIPTION
**Pull request by sduskis**
Moving adapter management into a new class called Adapters.

### CODE REVIEW

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java [hunk 1]
```diff
@@ -205,15 +206,11 @@ public BigtableBufferedMutator(
 
     this.host = options.getDataHost().toString();
 
-    PutAdapter putAdapter = new PutAdapter(configuration);
+    PutAdapter putAdapter =
+        new PutAdapter(configuration.getInt("hbase.client.keyvalue.maxsize", -1));
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: Perhaps wrap this logic in Adapters and call Adapters.createPutAdapter()?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableTable.java [hunk 2]
```diff
@@ -148,12 +135,8 @@ public BigtableTable(AbstractBigtableConnection bigtableConnection,
     this.tableName = tableName;
     this.options = options;
     this.client = client;
-    putAdapter = new PutAdapter(getConfiguration());
-    mutationAdapter = new MutationAdapter(
-        DELETE_ADAPTER,
-        putAdapter,
-        new UnsupportedOperationAdapter<Increment>("increment"),
-        new UnsupportedOperationAdapter<Append>("append"));
+    putAdapter = new PutAdapter(getConfiguration().getInt("hbase.client.keyvalue.maxsize", -1));
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableTable.java]: Adapters.createPutAdapter()?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.apache.hadoop.hbase.client.Append;
+import org.apache.hadoop.hbase.client.Increment;
+
+import com.google.cloud.bigtable.hbase.adapters.filters.FilterAdapter;
+
+/**
+ * Manages all Adapters
+ */
+public class Adapters {
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java]: nit: abstract class

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  public static final FilterAdapter FILTER_ADAPTER = FilterAdapter.buildAdapter();
+  public static final ScanAdapter SCAN_ADAPTER =  new ScanAdapter(FILTER_ADAPTER);
+  public static final BigtableResultScannerAdapter BIGTABLE_RESULT_SCAN_ADAPTER =
+      new BigtableResultScannerAdapter(ROW_ADAPTER);
+  public static final BigtableWhileMatchResultScannerAdapter
+      BIGTABLE_WHILE_MATCH_RESULT_RESULT_SCAN_ADAPTER =
+      new BigtableWhileMatchResultScannerAdapter(ROW_ADAPTER);
+  public static final GetAdapter GET_ADAPTER = new GetAdapter(SCAN_ADAPTER);
+
+  public static MutationAdapter createMutationsAdapter(PutAdapter putAdapter) {
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java]: Can PutAdapter be static as other adapters here?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/PutAdapter.java [hunk 5]
```diff
@@ -36,8 +35,8 @@
 public class PutAdapter implements OperationAdapter<Put, MutateRowRequest.Builder> {
   private int maxKeyValueSize;
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/PutAdapter.java]: private final int

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java (hunk 1 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: good idea.

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableTable.java (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableTable.java]: yup

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java (hunk 3 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java]: final with private constructor?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java (hunk 4 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java]: No... PutAdapter needs some configuration that's per Configuration..

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java (hunk 3 above)
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/Adapters.java]: yep

**Comment by sduskis** [comment]
Updated as per the comments.  PTAL

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**